Source/Scene/Model.js

```diff
@@ -1454,6 +1454,191 @@ define([
+                    // indicated by when an attribute points to a parameter with a
+                    // count property.
+                    //
+                    // https://github.com/KhronosGroup/glTF/issues/258
+
+                    var materialId = primitive.material;
+                    var material = materials[materialId];
+                    var techniqueId = material.technique;
+                    var technique = techniques[techniqueId];
+                    
+                    var programId = technique.program;
+                    if (programId === programName) {
+                        var primitiveAttributes = primitive.attributes;
+                        var attributeVarName;
+                        for (var attributeSemantic in primitiveAttributes) {
+                            if (primitiveAttributes.hasOwnProperty(attributeSemantic)) {
+                                attributeVarName = undefined;
+                                for (var attributeName in technique.parameters) {
+                                    if (technique.parameters.hasOwnProperty(attributeName)) {
+                                        if (defined(attributeVarName)) {
+                                            break;
+                                        }
+                                        if (technique.parameters[attributeName].semantic === attributeSemantic) {
+                                            for (var varName in technique.attributes) {
+                                                if (technique.attributes.hasOwnProperty(varName)) {
+                                                    if (technique.attributes[varName] === attributeName) {
+                                                        attributeVarName = varName;
+                                                        break;
+                                                    }
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                                // Skip if the attribute is not used by the material, e.g., because the asset was exported
+                                // with an attribute that wasn't used and the asset wasn't optimized.
+                                if (defined(attributeVarName)) {
+                                    var attributeId = primitiveAttributes[attributeSemantic];
+                                    var a = accessors[attributeId];
+                                    if (a.hasOwnProperty('extensions') && processedAccessors.indexOf(attributeId) < 0) {
+                                        var extensions = a.extensions;
+                                        if (extensions.hasOwnProperty('WEB3D_quantized_attributes')) {
+                                            // Only one of each semantic type can be decoded per shader
```

Comment:
You are correct that no primitive can have more than one attribute of each semantic type. 

This code addresses the case, as in the Cesium Milk Truck model, where two materials use the same technique and thus the same program. Then, two primitives with different accessors and different decode matrices use those materials. 

It creates a new, identical program and technique, then reassigns the primitive's material so that a separate shader is created.